### PR TITLE
Implemented additional data for keymap file and json parsing

### PR DESCRIPTION
**Pull request by 9gel**
You can now do:

`qmk compile keyboards/chonkerkeys/max/templates/chonkerkeys_max.json`

Please note the new JSON format there that will replace the old JSON format. This aligns more with the standard QMK format.

Also: the layout you specify in the JSON file will match what you see physically on the board because we did the "mirroring" (left to right) using the LAYOUT macros in max.h and original.h .

### CODE REVIEW

File: lib/python/qmk/keymap.py [hunk 1]
```diff
@@ -189,6 +189,43 @@ def generate_json(keymap, keyboard, layout, layers):
...
+                    layer_data = map(_remove_unsafe_chars, layer_data)
+                    layer_data_txt.append('\t[%s] = %s(%s)' % (layer_num, layout_macro, ', '.join(layer_data)))
+                else:
+                    layer_data_txt.append('\t' + _remove_unsafe_chars(layer_data))
+            data_name_txt = '__REPLACE_%s__' % _remove_unsafe_chars(data_name)
+            new_keymap = new_keymap.replace(data_name_txt, '\n'.join(layer_data_txt))
+
+    if keymap_json.get('layer_custom_keys'):
+        layer_keys_txt = []
+        c_macro = keymap_json['layout'] + '_CUSTOM_KEYS'
```
Comment (roxk) [inline lib/python/qmk/keymap.py:217]: All LGTM, and I'm thinking if QMK maintainers would accept these changes.

Other `REPLACE_$blah` pattern looks LGTM as those are easily customizable by other keyboards as well. I'd assume layer count is also usable by other keyboards. So these 2 changes should be easier to be accepted by QMK maintainers.

But I'm not sure if this `_CUSTOM_KEYS_` key thing is extensible enough to be accepted by them 🤔 I'm thinking can we make a general enough routine that can be used by both `key_map` and `custom_keymap`. Can we make `CUSTOM` replaceable like those in `REPLACE_$blah`? So the new code here would try to match `$blah_keys` to `$BLAH_KEYS_GOES_HERE`, The exact naming does't have to be `$blah_keys` as I see `layer` is used for normal keymaps. Maybe `layer_$blah`?

**Comment by roxk** [comment]
LGTM, just one comment. See if you like the idea, and if so, would want to include in this PR. I can merge the PR even if you don't bother considering the idea yet.

File: lib/python/qmk/keymap.py (hunk 1 above)
Comment (9gel) [reply lib/python/qmk/keymap.py:217]: Good points. But let's get this working first before adjusting the code so they are acceptable to maintainers. If I was the maintainer, there are a few things I'd want to change before accepting the change. _CUSTOM_KEYS_ can be controversial - but we need to get it working now so let's go forward. We can discuss with maintainers later how best to make that work.

**Comment by roxk** [comment]
@9gel Do you still have more commits to come? I can merge now if not. Thanks!

**Comment by 9gel** [comment]
> 



> @9gel Do you still have more commits to come? I can merge now if not. Thanks!

No more and I'm done. Thanks!

**Comment by roxk** [comment]
I will test CI after merging and notify you if anything breaks :p

**Pull request merged**